java/vector/src/main/codegen/templates/ComplexWriters.java

```diff
@@ -120,11 +120,18 @@ public void write(Nullable${minor.class}Holder h) {
   }
   </#if>
 
-  <#if minor.class == "VarChar">
+  <#if minor.class?ends_with("VarChar")>
+  @Override
   public void write${minor.class}(${friendlyType} value) {
     vector.setSafe(idx(), value);
     vector.setValueCount(idx()+1);
   }
+
+  @Override
+  public void write${minor.class}(String value) {
+    vector.setSafe(idx(), new Text(value));
```

Comment:
It's small, but is it worth caching the Text object in the writer implementation? It seems from the Text API that that was the intent.

Reply:
That makes sense. It looks like it'd avoid an allocation if write was called with strings of decreasing size.